common/luaUtilities/team_transfer/context_factory.lua

```diff
@@ -0,0 +1,148 @@
+-- Context Factory with Closures
+-- Functions that capture repositories in closures for cleaner APIs
+
+local SharedEnums = VFS.Include("common/luaUtilities/team_transfer/shared_enums.lua")
+
+---@class ContextFactory
+---@field create fun(springRepo: ISpring): ContextFactory
+---@field policy fun(senderTeamID: number, receiverTeamID: number): PolicyContext
+---@field action fun(senderTeamId: number, receiverTeamId: number, transferCategory: string): PolicyActionContext
+---@field resource fun(senderTeamId: number, receiverTeamId: number, resourceType: string, amount: number, policyResult: ResourcePolicyResult): ResourceTransferPolicyContext
+---@field resourceTransfer fun(senderTeamId: number, receiverTeamId: number, resourceType: ResourceType, desiredAmount: number, policyResult: ResourcePolicyResult): ResourceTransferContext
+local ContextFactory = {}
+
+---Get team resources as unpacked table for easier handling
+---@param springRepo ISpring
+---@param teamID number
+---@param resourceType string
+---@return ResourceData Complete resource data for the team
+local function getTeamResourcesUnpacked(springRepo, teamID, resourceType)
+    local current, storage, pull, income, expense, share, sent, received = springRepo.GetTeamResources(teamID, resourceType)
+
+    -- Handle nil values (e.g., for teams that haven't initialized resources yet)
+    return {
+        current = current,
+        storage = storage,
+        pull = pull,
+        income = income,
+        expense = expense,
+        shareSlider = share,
+        sent = sent,
+        received = received
+    }
+end
+
```

Comment:
This file is a wee bit silly without the other types of Transfer, buuuut it's simple enough and just maps spring onto types for me. Sharing these types between the upstream branch is useful to me and this only happens in the synced policy building side.